### PR TITLE
Do not copy files on restart

### LINKED ISSUE #510: Compressed files are copied when restarting jobs
When restarting from an existing calculation, the new job copies ALL the files from the original one by default. I feel this would become extremely costly in terms of disk space. An example code:

```python
from pyiron import Project
pr = Project("vasp_testing")
pr.remove_jobs_silently(True)
basis = pr.create_ase_bulk("Al")
job = pr.create_job("Vasp", "test")
job.structure = basis
job.run()
job_new = job.restart("test_restart")
job_new.run()
job_new.list_files()
```
 produces the output:

```
>>> ['CHGCAR', 'CONTCAR', 'test.tar.bz2', 'test_restart.tar.bz2']
```

'test.tar.bz2' should not be copied IMO

### PR DESCRIPTION
**Pull request by sudarsan-surendralal**
See linked issue

### CODE REVIEW

**Comment by pmrv** [comment]
We still want to copy certain files, though, like `CONTCAR`, right?  Is that taken care of separately?

**Comment by sudarsan-surendralal** [comment]
> 
> 
> We still want to copy certain files, though, like `CONTCAR`, right? Is that taken care of separately?

Yes, for that we need to use `restart_file_list` and/or  `restart_file_dict`.

File: pyiron_base/_tests.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 import unittest
 from os.path import split, join
 from os import remove
+import posixpath
```
Comment (jan-janssen) [inline pyiron_base/_tests.py]: Just for consistency is there a reason why `os.path.join()` is not sufficient?

**Comment by jan-janssen** [comment]
Just for me to understand this correctly, this behavior was introduced in https://github.com/pyiron/pyiron_base/pull/349 right?

File: pyiron_base/_tests.py (hunk 1 above)
Comment (sudarsan-surendralal) [reply pyiron_base/_tests.py:13]: > 
> 
> Just for consistency is there a reason why `os.path.join()` is not sufficient?

The `posixpath.join` method is [platform agnostic](https://www.georgevreilly.com/blog/2016/07/12/PythonJoiningUrlsWithPosixpathJoin.html) which is important esp for windows.

Comment (sudarsan-surendralal) [reply pyiron_base/_tests.py]: I guess `posixpath.join` is platform independent ([source](https://www.georgevreilly.com/blog/2016/07/12/PythonJoiningUrlsWithPosixpathJoin.html)) so we should use this.

**Comment by sudarsan-surendralal** [comment]
> 
> 
> Just for me to understand this correctly, this behavior was introduced in #349 right?

Yes and it worked well. But I guess having the new parameter `copy_files` with `True` as the default meant that the `restart` function which called this automatically copied all files (once again emphasizing the need for more tests on `pyiron_base`).

File: pyiron_base/_tests.py (hunk 1 above)
Comment (pmrv) [reply pyiron_base/_tests.py]: `os.path.join` should be doing the correct™ thing depending on the platform.

Comment (jan-janssen) [reply pyiron_base/_tests.py]: > I guess `posixpath.join` is platform independent ([source](https://www.georgevreilly.com/blog/2016/07/12/PythonJoiningUrlsWithPosixpathJoin.html)) so we should use this.

The link talks about URL not file paths. `posixpath` always gives you `/` while `os.path` gives `/` on linux and Mac OS X and `\` on windows. We want to use `posixpath` for all HDF5 internal path but for file path I would suggest to use `os.path.join` to be more consistent with the python standards.

**Comment by sudarsan-surendralal** [comment]
It seems that the output also seems to be copied during the restart. So I'm converting this to draft till I figure out what's going on. 

Edit: False alarm, everything seems okay! Do let me know if any further changes are required @jan-janssen @pmrv

**Comment by sudarsan-surendralal** [comment]
@niklassiemer @muh-hassani Can you make a new release with these changes (also for the cluster)?

**Comment by niklassiemer** [comment]
Sure, we will trigger a new release with minor (not only patch level) version bump once we fixed the current problems with the hdf and db connection bugs we have in our master.

**Pull request merged**